samples/performance/async_compute/async_compute.cpp

```diff
@@ -0,0 +1,809 @@
+/* Copyright (c) 2021, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "async_compute.h"
+
+#include "api_vulkan_sample.h"
+#include "common/vk_common.h"
+#include "gltf_loader.h"
+#include "gui.h"
+#include "platform/filesystem.h"
+#include "platform/platform.h"
+#include "scene_graph/components/orthographic_camera.h"
+#include "stats/stats.h"
+
+AsyncComputeSample::AsyncComputeSample()
+{
+	auto &config = get_configuration();
+
+	config.insert<vkb::BoolSetting>(0, async_enabled, false);
+	config.insert<vkb::BoolSetting>(1, async_enabled, true);
+	config.insert<vkb::BoolSetting>(0, rotate_shadows, false);
+	config.insert<vkb::BoolSetting>(1, rotate_shadows, true);
+	config.insert<vkb::BoolSetting>(0, double_buffer_hdr_frames, false);
+	config.insert<vkb::BoolSetting>(1, double_buffer_hdr_frames, true);
+}
+
+void AsyncComputeSample::draw_gui()
+{
+	gui->show_options_window(
+	    /* body = */ [this]() {
+		    ImGui::Checkbox("Enable async queues", &async_enabled);
+		    ImGui::Checkbox("Double buffer HDR", &double_buffer_hdr_frames);
+		    ImGui::Checkbox("Rotate shadows", &rotate_shadows);
+	    },
+	    /* lines = */ 3);
+}
+
+static VkExtent3D downsample_extent(const VkExtent3D &extent, uint32_t level)
+{
+	return {
+	    std::max(1u, extent.width >> level),
+	    std::max(1u, extent.height >> level),
+	    std::max(1u, extent.depth >> level)};
+}
+
+void AsyncComputeSample::prepare_render_targets()
+{
+	// To make this sample demanding enough to saturate mobile devices, use 4K.
```

Comment:
Using a size here that is larger than the swapchain causes significant performance issues on some of our devices (which are designed to be very small and low power). You noted that this was to allow comparing of cross-device performance, but I don't believe that's the aim of these performance samples; they allow you to compare various options on a single device.

Reply:
I hit vsync on the devices I test on unless I really push it. The intention is to be able to see a difference when toggling the options.

Reply:
Samples based on the ARM framework are always locked to vsync. We discussed this on the last call and hopefully have fix for that soon ;)

Reply:
> The sample is tagged Arm as well, FWIW

I'm not sure why it's tagged Arm. What here is Arm specific?

Reply:
It was tagged like that since it demonstrates performance differences on Mali. It's somewhat implementation specific how this works.

Reply:
Sure, but that's true of many of the performance samples, which aren't tagged as Arm specific.
My understanding was that the tag was for things that really are vendor specific.

Reply:
Fixed.

Reply:
Thanks.